### PR TITLE
fix(adapter-node): merge sourcemaps

### PR DESCRIPTION
**Pull request by gtm-nayan**
This targets the branch for #10041.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Comment by benmccann** [comment]
I didn't look at this one yet, but thought I'd give you a heads up that `lint` is failing

File: packages/adapter-node/index.js [hunk 1]
```diff
@@ -111,3 +114,27 @@ export default function (opts = {}) {
...
 	};
 }
+
+/**
+ * Load sourcemaps for files in the tmp directory so that the final ones
+ * point to the original source files, instead of the generated files in outDir.
+ *
+ * @param {string} tmp
+ * @returns {import('rollup').Plugin}
+ * */
```
Comment (benmccann) [inline packages/adapter-node/index.js:124]: The last line looks a bit malformed. Also, I just did a quick search with `grep -B 1 -r @param packages/kit/src/` and it looks like our convention is to not have a blank line
```suggestion
 * @param {string} tmp
 * @returns {import('rollup').Plugin}
 */
```

File: packages/adapter-node/index.js [hunk 2]
```diff
@@ -111,3 +114,26 @@ export default function (opts = {}) {
...
+function merge_sourcemap_plugin(tmp) {
+	const should_process_sourcemaps = createFilter(`${normalizePath(tmp)}/**/*.js`);
+
+	return {
+		name: 'adapter-node-sourcemap-loader',
+		async load(id) {
+			if (!should_process_sourcemaps(id)) return;
+			if (!existsSync(`${id}.map`)) return;
+			const [code, map] = await Promise.all([
+				readFile(id, 'utf-8'),
```
Comment (dummdidumm) [inline packages/adapter-node/index.js:133]: why is this using `readFile` when other places use `readFileSync`? Is this really an observable performance difference?

Comment (benmccann) [reply packages/adapter-node/index.js:133]: this one is reading two files in parallel, so I think `readFile` would be more performant here, so if we want to standardize I think I'd rather use `readFile` than `readFileSync`

**Review by dummdidumm — APPROVED** [review]

**Pull request merged**